tools/fs/files.js

```diff
@@ -926,8 +940,13 @@ files.createTarGzStream = function (dirPath, options) {
       // setting it in an 'entry' handler is the same strategy that npm
       // does, so we do that here too.
       if (entry.type === "Directory") {
-        entry.mode = (entry.mode || entry.props.mode) | 0o500;
-        entry.props.mode = entry.mode;
+        entry.props.mode = addExecBitWhenReadBitPresent(entry.props.mode);
```

Comment:
Why is the `| 0o500` no longer necessary? In other words, why not something like this:
```js
entry.props.mode = addExecBitWhenReadBitPresent(
  (entry.mode || entry.props.mode) | 0o500
);
```